src/components/useMultiStepForm.ts

```diff
@@ -0,0 +1,34 @@
+import { ReactElement, useState } from "react";
+
+export function useMultiStepForm(steps: ReactElement[]) {
+  const [currentStepIndex, setCurrentStepIndex] = useState(0);
+
+  function next() {
+    setCurrentStepIndex((i: number) => {
+      if (i >= steps.length - 1) return i;
+      return i + 1;
+    });
+  }
+
+  function back() {
+    setCurrentStepIndex((i: number) => {
+      if (i <= 0) return i;
+      return i - 1;
+    });
+  }
+
+  function goTo(index: number) {
+    setCurrentStepIndex(index);
```

Comment:
Is there a chance that an incorrect number could be added causing the goTo fn to step out of the valid range. Like if I accidentally put in 4?
```suggestion
    if (index < 0 || index >= steps.length) return;
    setCurrentStepIndex(index);
```